code/_core/obj/item/fishing/rod.dm

```diff
@@ -0,0 +1,282 @@
+/obj/item/fishing/rod
+	name = "fishing rod"
+	drop_sound = 'sound/items/drop/scrap.ogg'
+	size = SIZE_2
+	weight = 2
+	var/lavaproof = 0
+	var/lcolor = "#FFFFFF"
+	var/nice_bait = 0
+	var/break_on_failure = 0
+	var/break_on_success = 0
+	var/win = 0
+	var/win_time = 0
+	var/timemin = 3
+	var/timemax = 9
+	var/react = 9
+	var/bobsprite = null
+	var/catchsound = "sound/ui/message_ping.ogg"
+	var/fishing = 0
+	var/fishingzone = 0
+	var/wished_fishingzone = 0
+	var/mintime = 0
+	var/maxtime = 0
+	var/list/obj/item/effrewards = list()
+	var/list/obj/item/wekrewards = list()
+	var/fish_loc
+
+	var/obj/item/fishing/lure
+	var/obj/item/fishing/line
+	var/obj/item/fishing/bait
+
+/obj/item/fishing/rod/save_item_data(var/save_inventory = TRUE)
+	. = ..()
+	SAVEATOM("lure")
+	SAVEATOM("line")
+	SAVEATOM("bait")
+	return .
+
+/obj/item/fishing/rod/load_item_data_pre(var/mob/living/advanced/player/P,var/list/object_data)
+	. = ..()
+	LOADATOM("lure")
+	LOADATOM("line")
+	LOADATOM("bait")
+	update_rod_stats()
+	return .
+
+/obj/item/fishing/rod/proc/get_line()
+	return line
+
+/obj/item/fishing/rod/proc/get_lure()
+	return lure
+
+/obj/item/fishing/rod/proc/get_bait()
+	return bait
+
+/obj/item/fishing/rod/proc/update_rod_stats()

```

Comment:
I think instead of storing all the variables on the fishing rod, you could just grab the fishing components (Line, Lure, Bait) as it shows up and needing to update the stats won't be needed.

Reply:
did a good chunk of that for next commit